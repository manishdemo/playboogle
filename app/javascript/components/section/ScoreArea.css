table {
    border-collapse: collapse;
    border: 1px solid black;
    width: 80%;
    margin-left: 30px;

}

th {
    width: 40%;
}

td {
    width: 50%;
    text-align: center;
    alignment: center;
    align-items: center;
}

tr:nth-child(even) {
    background-color: #d3f7ad;
}

.total-score {
    text-align: center;
}

.score-area {
    background: #95bb76;
    width: 48%;
    float: left;
    padding: 10px;
    height: 350px;
    border-bottom-right-radius: 25px;
    border-top-right-radius: 25px;
}