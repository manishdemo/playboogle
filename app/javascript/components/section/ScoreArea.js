import React from "react";
import "./ScoreArea.css";

export class ScoreArea extends React.Component {
    render() {
        return (
            <div className="score-area">
                <h1 align={"center"}> Score Board </h1>
                <table>
                    <tr>
                        <th>Word</th>
                        <th>Point</th>
                    </tr>
                    <tr>
                        <td>Apple</td>
                        <td>2</td>
                    </tr>
                    <tr>
                        <td>Ball</td>
                        <td>1</td>
                    </tr>
                    <tr>
                        <td>Cat</td>
                        <td>1</td>
                    </tr>
                </table>
                <br/>
                <br/>
                <div className={"total-score"}>
                    Total Score: 4
                </div>
            </div>
        )
    }
}

export default ScoreArea
