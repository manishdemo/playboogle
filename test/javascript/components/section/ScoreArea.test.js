import React from "react";

import ScoreArea from "../../../../app/javascript/components/section/ScoreArea";
import {describe, expect, it} from "@jest/globals";
import {shallow, render} from "enzyme";

describe('ScoreArea', () => {
    it('should render correctly', () => {
        const history = {
            "cat": 1,
            "dog":1,
            "zebra": 2
        };

        const component = shallow(<ScoreArea history={history} /> );
        expect(component).toMatchSnapshot();
    });

    it('should have number of displayed words as per the provided input ', () => {
        const history = {
            "cat": 1,
            "dog":1,
            "zebra": 2
        };

        const wrapper = render(<ScoreArea history={history} /> );
        expect(wrapper.find('tr')).toHaveLength(Object.keys(history).length+1);
    });

});