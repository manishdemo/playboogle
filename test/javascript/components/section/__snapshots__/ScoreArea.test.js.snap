// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScoreArea should render correctly 1`] = `
<ScoreArea
  history={
    Object {
      "cat": 1,
      "dog": 1,
      "zebra": 2,
    }
  }
  innerRef={null}
/>
`;
